src/utils/buildMetadata.ts

```diff
@@ -166,13 +153,218 @@ export function buildMetadata(sheets: Sheet[], defaultCC: string) {
         optionSets,
         trackedEntityAttributes,
         trackedEntityTypes,
-        //programs,
+        programSections: buildprogramSections(sheets),
+        programs: buildPrograms(sheets),
+        programStages: buildProgramStages(sheets),
+        programStageSections: buildProgramStageSections(sheets),
         programRules: buildProgramRules(sheets),
         programRuleActions: buildProgramRuleActions(sheets),
         programRuleVariables: buildProgramRuleVariables(sheets),
     };
 }
 
+function buildPrograms(sheets: Sheet[]) {
+    const get = (name: string) => getItems(sheets, name);
+
+    const programs = get("programs");
+    const programSectionsData = get("programSections");
+    const programStagesData = get("programStages");
+    const trackedEntityTypes = get("trackedEntityTypes");
+    const categoryCombos = get("categoryCombos");
+
+    return programs.map(program => {
+        const trackedEntityType = {
+            id: getByName(trackedEntityTypes, program.trackedEntityType)?.id
+        };
+
+        const programStages = programStagesData
+            .filter((programStages) => {
+                return programStages?.program === program.name;
+            }).map(({ id }) => ({ id }));
+
+        const categoryCombo = {
+            id: getByName(categoryCombos, program.categoryCombo)?.id ?? process.env.DEFAULT_CATEGORY_COMBO_ID
+        };
+
```

Comment:
If we use `let data = ...` as mentioned in another comment, an easier alternative would be:

```
replaceById(data, "categoryCombo", categoryCombos);
```

I'm not sure if we need the defaultCategoryCombo in this case, but if we do, I think it'd be better to receive it as a parameter (like `defaultCC` in `buildMetaData()`) and then use a modified version of `replaceById()` that could look like this:

```
function replaceById(data: MetadataItem, key: string, items: MetadataItem[], defaultId=undefined) {
    if (key in data) {
        const name = data[key];
        const item = getByName(items, name);
        data[key] = { id: item.id };
    }
    else if (defaultId) {
        data[key] = { id: defaultId };
    }
}
```

Reply:
After testing it, defaultCategoryCombo gets assigned by DHIS2 if categoryCombo is not present.
Empty categoryCombo ("categoryCombo": {} or "categoryCombo": {id: ""}) results in an import error.
So, no need for the fallback value.
Changing it to replaceById()

Reply:
So either i modify `replaceById()` function or i address it in `buildPrograms()`.

`replaceById()` option:
```
function replaceById(data: MetadataItem, key: string, items: MetadataItem[], defaultId?: boolean) {
    if (key in data) {
        const name = data[key];
        const item = getByName(items, name);
        data[key] = { id: item.id };
    }
    else if (defaultId) {
        data[key] = { id: defaultId };
    }
}
```
I think that `defaultId?: boolean` works better as an optional flag to modify the function behavior than passing an undefined.

`buildPrograms()` option:
```
const categoryComboID = getByName(categoryCombos, program.categoryCombo)?.id;
const categoryCombo = categoryComboID !== undefined ? { id: categoryComboID } : undefined;
```

Reply:
Just to note it here, I understand that this is solved by using with the changes in `development` too (which will not add an empty field `categoryCombo` if it is not there). And this way we do not need to treat the categoryCombo in any special way, right?

Reply:
That's right.